Fix icons on App page in FF

rullzer opened the pull request:
For #900
- [x] Filter not working
- [x] No scaling

Once done:
- [x] Check Chrome
- [x] Check FF
- [x] Check IE/Edge
- [x] Check Safari
- [x] Check opera

rullzer committed:
Filter can't be named invert in FF

MorrisJobke committed:
Introduce weird hack for Firefox

* everything is 4 times as big (doubled width and doubled height)

blizzz commented on settings/js/apps.js:
```diff
@@ -234,6 +234,11 @@ OC.Settings.Apps = OC.Settings.Apps || {
 	imageUrl : function (url, appfromstore) {
 		var img = '<svg width="72" height="72" viewBox="0 0 72 72">';
+		// weird hack because in Firefox everything is 4 times as big (doubled in width and doubled in height)
+		// TODO: replace this with a proper solution
```
can we have todos as issues instead, pls?

MorrisJobke replied:
I will create it

MorrisJobke replied:
#908

rullzer commented:
Fine by me for now :+1:

rullzer committed:
Remove hack

rullzer committed:
Add viewBox to app icons

If we don't add a viewbox a lot of browsers will just say NOPE!!! when
we ask them to scale.

rullzer commented:
It seems it is because a lot of icons are missing a `viewBox` attribute. So browsers just say sorry but I can't scale that.
I pushed some updated icons to test.

MorrisJobke committed:
Empty tags are not allowed for image and feColorMatrix in IE11 and below

MorrisJobke committed:
Fix null reference in IE

MorrisJobke commented:
I tested this now in Edge, IE11, IE10, Safari and it works just fine 👍

MorrisJobke commented:
@karlitschek We need this backported (and it would be best to have this in before the RC - looking towards @LukasReschke 😉 )

MorrisJobke commented:
I tested this with Opera too and it works there :)

karlitschek commented:
nice. please backport

LukasReschke commented:
LGTM and world doesn't explode 👍

MorrisJobke commented:
@karlitschek for all the backports above 🙈 It's mostly only adding a `viewBox="0 0 32 32"` to the SVGs because otherwise FF and IE doesn't scale SVGs at all. In other browsers this is the default behavior.

Then the app management looks nice for all shipped apps.
